routes/approvals.js

```diff
@@ -40,11 +40,36 @@ approvals.getApprovals = function(app, res, loggedInUserId) {
     var userInfo = users.loadUser(app, loggedInUserId);
     if (!userInfo)
         return res.status(403).jsonp({ message: 'Not allowed' });
-    if (!userInfo.admin)
+    if (!userInfo.admin && !userInfo.approver)
         return res.status(403).jsonp({ message: 'Not allowed' });
-    
+
     var approvalInfos = approvals.loadApprovals(app);
+    approvalInfos = approvalInfos.filter(function(approval) {
+      if(userInfo.admin) return true; //Show all approvals for admin
+      if(!userInfo.groups) return false; //Approver is not attached to a group (un-likely)
+      if(!approval.api.requiredGroup) return false; //Api is not attached any group
+
+      var groups = utils.loadGroups(app);
+      groups = groups.groups;
+      if(!groups) return false;
+      var usrGroupsHash = {};
+      for(var i=0; i< userInfo.groups.length; i++){
+        usrGroupsHash[userInfo.groups[i]] = userInfo.groups[i];
+      }
+      for(var i=0; i< groups.length; i++){
+        if(groups[i].id==usrGroupsHash[groups[i].id]){
+          var alt_ids = groups[i].alt_ids;
```

Comment:
This is a little unclear; what are using the `alt_ids` for? They are intended for mapping external group IDs (e.g. from an AD or similar) to internal wicked groups. What's the use case for mixing them in here?